src/lib/builders/date-picker/create.ts

```diff
@@ -131,6 +131,30 @@ export function createDatePicker(props?: CreateDatePickerProps) {
 		calendar.options.maxValue.set($maxValue);
 	});
 
+	if (options.numberOfMonths != undefined) {
```

Comment:
Why do we need this conditional here but not for the others? This _should_ always be defined considering we're creating a store for each prop.

Does this not work if its the same as the ones above it?

Reply:
Ahh I see, makes sense!

Reply:
I don't have a super clear idea of the implications creating the default all the times means that there's an extra step that always need to happen (the synchronization) even if someone doesn't pass a default value for it (currently they interact directly with the calendar store in that case)

Reply:
Oh and btw this was sort of a question...like: what would you like to do? 😁

Reply:
I think for now we apply the defaults to keep it in-line with the rest of it in the event we add more props in the future or change a specific prop for the calendar that only would apply as the default on the date picker. I should have had this in there from the beginning the more that I think about it!

Reply:
Gotcha boss...will work on this tomorrow.

Reply:
Uh now that I think about it...this might be a change in the typings (I have to verify) and thus a breaking change (number of months is possibly undefined even in the front facing builder). Could that be considered a breaking change?

Reply:
@huntabyte i made the change and i can confirm that it will change the frontfacing types. Technically this is a breaking change but should be a small one since something that could've been undefined it's now never undefined. What do we want to do about this?

cc. @TGlide

Reply:
Breaking changes are fine, we just mark them as a minor patch.

Reply:
Gotcha...just pushed the changes then 😄

Reply:
Do you mean return type differences? I don't see a change to the `types.ts` file here! @paoloricciuti

Reply:
> Do you mean return type differences? I don't see a change to the `types.ts` file here! @paoloricciuti

Yeah return type differences...if you use `createDatePicker` and destructure `numberOfMonths` now is of type `number` before it was `number | undefined`